bumble/device.py

```diff
@@ -2808,11 +2808,23 @@ async def get_pin_code():
                 )
 
                 if pin_code is not None:
-                    pin_code = bytes(str(pin_code).zfill(6))
+                    # TODO: make pin code input more flexible,
+                    # so far only numbers are allowed
+                    pin_code = bytes(str(pin_code), encoding='utf8')
+                    pin_code_len = len(pin_code)
+                    if pin_code_len >= 0x10:
+                        logger.warning(
+                            'pin_code is longer than 16 bytes, truncated to 16'
+                        )
+                        pin_code_len = 0x10
+                        pin_code = pin_code[:16]
+                    else:
+                        pin_code = pin_code + b'0' * (16 - pin_code_len)
```

Comment:
You shouldn't need to manually pad the field here, the HCI object serializer automatically truncates/pads byte array fields when they are fixed length.

Reply:
Updated.